src/crate/client/test_http.py

```diff
@@ -362,10 +360,59 @@ def test_no_params(self):
 
 
 class RequestsCaBundleTest(TestCase):
-
     def test_open_client(self):
         os.environ["REQUESTS_CA_BUNDLE"] = "/etc/ssl/certs/ca-certificates.crt"
         try:
             Client('http://127.0.0.1:4200')
         except ProgrammingError:
             self.fail("HTTP not working with REQUESTS_CA_BUNDLE")
+
+
+class RetryRequestHandler(BaseHTTPServer.BaseHTTPRequestHandler):
+    """
+    http handler for use with BaseHTTPServer
+
+    counts request made
+    """
+
+    def do_GET(self):
+        self.server.SHARED['count'] += 1
+
+    do_POST = do_PUT = do_DELETE = do_HEAD = do_GET
+
+
+class TestingHTTPServer(BaseHTTPServer.HTTPServer):
+    """
+    http server providing a shared dict
+    """
+    manager = multiprocessing.Manager()
+    SHARED = manager.dict()
+    SHARED['count'] = 0
+
+    def run_server(server_address):
+        TestingHTTPServer(server_address, RetryRequestHandler).serve_forever()
+
+
+class RetryOnTimeoutServerTest(TestCase):
+    server_address = ("127.0.0.1", 65535)
+
+    def __init__(self, *args, **kwargs):
+        super(RetryOnTimeoutServerTest, self).__init__(*args, **kwargs)
+        self.server_process = Process(target=TestingHTTPServer.run_server, args=(self.server_address,))
+
+    def setUp(self):
+        super(RetryOnTimeoutServerTest, self).setUp()
```

Comment:
it is not really required to call setUp or tearDown of TestCase, the default implementations do nothing